### PR TITLE
Clarify terminology around aggregations

### LINKED ISSUE #1405: Aggregations are confusing
An `m.replace` event is aggregated by a server to (sometimes) create a bundle: none of these words are intuitive. 

See https://spec.matrix.org/v1.4/client-server-api/#server-side-aggregation-of-mreplace-relationships and related sections.

### PR DESCRIPTION
**Pull request by richvdh**
I've done my best to remove the word "bundle", because I feel like it causes more confusion than clarification. Instead I have favoured "aggregated child events" which I think is clearer.

Some general clarification around these parts of the spec.

Hopefully, fixes #1405 

Preview: https://pr1424--matrix-spec-previews.netlify.app

### CODE REVIEW

File: data/api/client-server/threads_list.yaml [hunk 1]
```diff
@@ -32,8 +32,7 @@ paths:
       x-addedInMatrixVersion: "1.4"
       summary: Retrieve a list of threads in a room, with optional filters.
       description: |-
-        Paginates over the thread roots in a room, ordered by the `latest_event` of each thread root
-        in its bundle.
+        Paginates over the thread roots in a room.
```
Comment (turt2live) [inline data/api/client-server/threads_list.yaml:35]: The description appears to have lost some context here?

File: meta/documentation_style.rst [hunk 2]
```diff
@@ -44,33 +44,36 @@ Stylistic notes
...
-*name* of a property - and be mindful of the scope for confusion with cryptographic
-keys.
+* When talking about properties in JSON objects, prefer the word "property" to "field",
+  "member", or various other alternatives. For example: "this property will be set to
+  X if ...". Also avoid the term "key" unless you are specifically talking about the
+  *name* of a property - and be mindful of the scope for confusion with cryptographic
+  keys.
+
+* "i.e." (*id est*) is an abbreviation and hence is written with two full
+  stops. So is "e.g." (*exempli gratia*).
```
Comment (turt2live) [inline meta/documentation_style.rst:76]: This should probably be its own PR so it doesn't get buried in our future search history.

Also I'm not currently convinced it's a thing we [should] do - given the first point is that the language should be unambiguous, shouldn't we be using complete words/sentences instead? ("For example", "As demonstrated below", "..., such as when", etc)

**Review by turt2live — COMMENTED** [review]
Generally this lgtm with a couple (hopefully easy) comments to resolve.

File: data/api/client-server/threads_list.yaml (hunk 1 above)
Comment (richvdh) [reply data/api/client-server/threads_list.yaml:35]: It seemed redundant to specify the sort order here as well as in the response.

File: meta/documentation_style.rst (hunk 2 above)
Comment (richvdh) [reply meta/documentation_style.rst:76]: I'll move it out of here.

File: data/api/client-server/threads_list.yaml (hunk 1 above)
Comment (turt2live) [reply data/api/client-server/threads_list.yaml:35]: it adds a bit of clarity without having to scroll past a whole request schema, though not all that bothered.

The `summary` and `description` should be switched if we are losing the text though: it's more awkward if the `summary` is longer than the `description`.

Comment (richvdh) [reply data/api/client-server/threads_list.yaml]: Have updated a bit more, with reference to https://github.com/matrix-org/matrix-spec/blob/main/meta/documentation_style.rst#openapi.

If you still don't like it, am happy to put it back how it was.

**Review by turt2live — APPROVED** [review]

**Pull request merged**